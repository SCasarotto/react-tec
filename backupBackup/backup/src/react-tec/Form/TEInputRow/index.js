import TEInputRow from './TEInputRow';

export default TEInputRow;
