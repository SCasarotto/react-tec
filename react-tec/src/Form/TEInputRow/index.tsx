import React from 'react'

import TERow, { TERowCustomProps } from '../TERow'
import TELabel, { TELabelCustomProps } from '../TELabel'
import TEInput, { TEInputProps } from '../TEInput'

export interface TEInputRowProps
	extends TERowCustomProps,
		TELabelCustomProps,
		TEInputProps {
	labelForKey: string
}

const TEInputRow: React.FC<TEInputRowProps> = (props) => {
	const {
		rowSize,
		last,
		className = '',
		title,
		type,
		onChange,
		disabled,
		value,
		required,
		labelForKey,
		...rest
	} = props

	return (
		<TERow
			rowSize={rowSize}
			last={last}
			className={`TEInputRow ${className}`}>
			<TELabel
				htmlFor={labelForKey}
				required={required}
				disabled={disabled}
				className="TEInputRowTitle">
				{title}
			</TELabel>
			<TEInput
				type={type}
				value={value}
				onChange={onChange}
				id={labelForKey}
				disabled={disabled}
				className="TEInputRowInput"
				{...rest}
			/>
		</TERow>
	)
}

export default TEInputRow
