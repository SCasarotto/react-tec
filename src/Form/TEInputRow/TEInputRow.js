//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import TERow from './../TERow'
import TELabel from './../TELabel'
import TEInput from './../TEInput'
import TETextarea from './../TETextarea'

const TEInputRow = (props) => {
	const {
		//Row
		size,
		last,
		className,

		//Label
		title,

		//Input
		type,
		onChange,
		disabled,
		value,

		//Both
		required,
		labelForKey,

		...rest
	} = props

	return (
		<TERow size={size} last={last} className={className}>
			<TELabel
				htmlFor={labelForKey}
				required={required}
				disabled={disabled}
				className='TEInputRowTitle'
			>
				{title}
			</TELabel>
			{type !== 'textarea' ? (
				<TEInput
					type={type}
					value={value}
					onChange={onChange}
					id={labelForKey}
					disabled={disabled}
					className='TEInputRowInput'
					{...rest}
				/>
			) : (
				<TETextarea
					value={value}
					onChange={onChange}
					id={labelForKey}
					disabled={disabled}
					className='TEInputRowInput'
					{...rest}
				/>
			)}
		</TERow>
	)
}

TEInputRow.propTypes = {
	//Row
	rowStyles: PropTypes.object,
	size: PropTypes.string,
	last: PropTypes.bool,

	//Input
	inputStyles: PropTypes.object,
	type: PropTypes.string,
	onChange: PropTypes.func,
	onChangeKey: PropTypes.string,
	disabled: PropTypes.bool,

	//Both
	value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
	labelForKey: PropTypes.string,
}

export default TEInputRow
