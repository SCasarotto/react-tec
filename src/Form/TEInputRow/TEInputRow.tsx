import React from 'react'

import { TERow, TERowCustomProps } from '../TERow'
import { TELabel, TELabelCustomProps } from '../TELabel'
import { TEInput, TEInputProps } from '../TEInput'

export interface TEInputRowProps extends TERowCustomProps, TELabelCustomProps, TEInputProps {
	labelForKey: string
}

export const TEInputRow = React.forwardRef<HTMLInputElement, TEInputRowProps>((props, ref) => {
	const { rowSize, last, className = '', title, disabled, required, labelForKey, ...rest } = props

	return (
		<TERow rowSize={rowSize} last={last} className={`TEInputRow ${className}`}>
			{title && (
				<TELabel
					htmlFor={labelForKey}
					required={required}
					disabled={disabled}
					className='TEInputRowTitle'
				>
					{title}
				</TELabel>
			)}
			<TEInput
				id={labelForKey}
				disabled={disabled}
				className='TEInputRowInput'
				ref={ref}
				{...rest}
			/>
		</TERow>
	)
})
