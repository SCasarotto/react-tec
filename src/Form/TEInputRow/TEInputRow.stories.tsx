import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEInputRow, TEInputRowProps } from './TEInputRow'

export default {
	title: 'Form/TEInputRow',
	component: TEInputRow,
	argTypes: {},
} as Meta

const Template: Story<TEInputRowProps> = (args) => <TEInputRow {...args} />

export const Default = Template.bind({})
Default.args = {
	title: 'My Input',
	labelForKey: 'SomeUniqueKey42',
	onChange: (e) => console.log(e.target.value),
}

export const Number = Template.bind({})
Number.args = {
	type: 'number',
	labelForKey: 'SomeUniqueKey666',
	title: 'My Number Input',
	onChange: (e) => console.log(e.target.value),
}
