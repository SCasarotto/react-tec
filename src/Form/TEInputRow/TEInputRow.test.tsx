import React from 'react'
import { render } from '@testing-library/react'
import { TEInputRow } from './TEInputRow'

describe('TEInputRow', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(<TEInputRow labelForKey='testTEInputRow' />)
		expect(asFragment()).toMatchSnapshot()
	})
})
