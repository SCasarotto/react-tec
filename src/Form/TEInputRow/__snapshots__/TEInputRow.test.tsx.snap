// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEInputRow matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  margin-bottom: 10px;
  text-align: left;
  box-sizing: border-box;
  padding-right: 0px;
  width: 100%;
}

.c1 {
  position: relative;
  display: block;
  width: 100%;
  font-size: 14px;
  padding: 10px;
  box-sizing: border-box;
  border-radius: 5px;
  margin: 0px;
  border: 1px solid;
  -webkit-transition: border-color 0.2s ease-in,box-shadow 0.2s ease-in;
  transition: border-color 0.2s ease-in,box-shadow 0.2s ease-in;
}

.c1:hover,
.c1:active {
  border: 1px solid;
  box-shadow: 0 0 0 1px inset;
}

.c1:disabled {
  border: 1px solid;
  box-shadow: none;
  cursor: not-allowed;
}

<div
    class="c0 TERow TEInputRow "
  >
    <input
      class="c1 TEInput TEInputRowInput"
      id="testTEInputRow"
    />
  </div>
</DocumentFragment>
`;
